tests/mc_simulator/test_mc_simulator.py

```diff
@@ -846,3 +891,96 @@ def test_MantarrayMcSimulator__does_not_send_status_beacon_while_rebooting(
     invoke_process_run_and_check_errors(simulator)
     actual_beacon_packet = simulator.read(size=STATUS_BEACON_SIZE_BYTES)
     assert actual_beacon_packet == bytes(0)
+
+
+def test_MantarrayMcSimulator__allows_metadata_to_be_set_through_testing_queue(
+    mantarray_mc_simulator,
+):
+    simulator = mantarray_mc_simulator["simulator"]
+    testing_queue = mantarray_mc_simulator["testing_queue"]
+
+    expected_metadata = {
+        TOTAL_WORKING_HOURS_UUID: 0,
+        MANTARRAY_NICKNAME_UUID: "New Nickname",
+    }
+    test_command = {"command": "set_metadata", "metadata_values": expected_metadata}
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_command, testing_queue
+    )
+    invoke_process_run_and_check_errors(simulator)
+
+    actual_metadata = simulator.get_metadata_dict()
+    # Check that expected values are updated
+    assert actual_metadata[TOTAL_WORKING_HOURS_UUID.bytes] == convert_to_metadata_bytes(
+        expected_metadata[TOTAL_WORKING_HOURS_UUID]
+    )
+    assert actual_metadata[MANTARRAY_NICKNAME_UUID.bytes] == convert_to_metadata_bytes(
+        expected_metadata[MANTARRAY_NICKNAME_UUID]
+    )
+    # Check that at least one other value is not changed
+    assert actual_metadata[
+        MANTARRAY_SERIAL_NUMBER_UUID.bytes
+    ] == convert_to_metadata_bytes(MantarrayMcSimulator.default_mantarray_serial_number)
+
+
+def test_MantarrayMcSimulator__allows_mantarray_nickname_to_be_set_by_command_received_from_pc__and_sends_correct_response(
+    mantarray_mc_simulator_no_beacon, mocker
+):
+    simulator = mantarray_mc_simulator_no_beacon["simulator"]
+    spied_get_cms_since_init = mocker.spy(simulator, "get_cms_since_init")
+
+    expected_nickname = "Newer Nickname"
+    dummy_timestamp = 0
+    set_nickname_command = create_data_packet(
+        dummy_timestamp,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_SET_NICKNAME_PACKET_TYPE,
+        convert_to_metadata_bytes(expected_nickname),
+    )
+    simulator.write(set_nickname_command)
+    invoke_process_run_and_check_errors(simulator)
+
+    # Check that nickname is updated
+    actual_metadata = simulator.get_metadata_dict()
+    assert actual_metadata[MANTARRAY_NICKNAME_UUID.bytes] == convert_to_metadata_bytes(
+        expected_nickname
+    )
+    # Check that correct response is sent
+    expected_response = create_data_packet(
+        spied_get_cms_since_init.spy_return,
```

Comment:
in the future, it may be good to figure out a way to test this that doesn't require checking the timestamp as well (unless you think that's critical to this unit of functionality)--as I imagine this may come up more frequently as new packet types are added

Reply:
I agree this is a good idea. In some cases we definitely want the timestamp, but in others we are mostly focused on testing the module ID, packet type, and main body of the data packet and possibly the checksum. If we think only one checksum related test is sufficient per module (one for packets sent from mc_comm and one for packets sent from the simulator) then we could definitely ignore the timestamp in all other tests